src/coreclr/src/vm/excep.cpp

```diff
@@ -3557,10 +3557,15 @@ BOOL StackTraceInfo::AppendElement(BOOL bAllowAllocMem, UINT_PTR currentIP, UINT
         // the current exception). Hence, set the corresponding flag to FALSE.
         pStackTraceElem->fIsLastFrameFromForeignStackTrace = FALSE;
 
+        // We can assume that IP points to an the instruction after [call IL_Throw] when these conditions are met:
+        //  1. !pCf->HasFaulted() - It wasn't a "hardware" exception (Access violation, dev by 0, etc.)
+        //  2. !pCf->IsIPadjusted() - It hasn't been previously adjusted to point to the call (i.e. like [call IL_Throw], etc.)
+        pStackTraceElem->fAdjustOffset = !pCf->HasFaulted() && !pCf->IsIPadjusted();
+
         // This is a workaround to fix the generation of stack traces from exception objects so that
         // they point to the line that actually generated the exception instead of the line
         // following.
-        if (!(pCf->HasFaulted() || pCf->IsIPadjusted()) && pStackTraceElem->ip != 0)
+        if (!pStackTraceElem->fAdjustOffset && pStackTraceElem->ip != 0)
```

Comment:
This logic appears to have been negated?
```
!fAdjustOffset == 
!(!pCf->HasFaulted() && !pCf->IsIPadjusted()) ==
 ( pCf->HasFaulted() ||  pCf->IsIPadjusted())
```

Reply:
Not sure what I was thinking. It is simple Boolean logic.